### PR TITLE
Merge shuffle changes from dev into master

### PR DESCRIPTION
**Pull request by jameelbrochu**
Merge changes from Sprint 2 including shuffle deck functionality.

### CODE REVIEW

File: app/src/androidTest/java/org/liberty/android/fantastischmemo/test/db/CardDaoTest.java [hunk 1]
```diff
@@ -17,7 +17,9 @@
 import java.util.Date;
 import java.util.List;
 
-import static org.junit.Assert.assertEquals;
+import static junit.framework.Assert.assertEquals;
```
Comment (rekandiah) [inline app/src/androidTest/java/org/liberty/android/fantastischmemo/test/db/CardDaoTest.java]: have third party imports such as junit before the java imports

File: app/src/main/java/org/liberty/android/fantastischmemo/dao/DeckDaoImpl.java [hunk 2]
```diff
@@ -1,12 +1,17 @@
 package org.liberty.android.fantastischmemo.dao;
 
 import com.j256.ormlite.dao.BaseDaoImpl;
+import com.j256.ormlite.stmt.PreparedQuery;
+import com.j256.ormlite.stmt.QueryBuilder;
 import com.j256.ormlite.support.ConnectionSource;
 import com.j256.ormlite.table.DatabaseTableConfig;
 
+import org.liberty.android.fantastischmemo.entity.Card;
+import org.liberty.android.fantastischmemo.entity.Category;
```
Comment (rekandiah) [inline app/src/main/java/org/liberty/android/fantastischmemo/dao/DeckDaoImpl.java]: android imports at the top plis

File: app/src/main/java/org/liberty/android/fantastischmemo/dao/CardDaoImpl.java [hunk 3]
```diff
@@ -858,5 +869,67 @@ private void maintainOrdinal() throws SQLException {
             throw new RuntimeException(e);
         }
     }
-}
+
+    public void shuffleCards(){
+        List<Card> cards = queryForAll();
+
+        Card[] cs = cards.toArray(new Card[cards.size()]);
```
Comment (rekandiah) [inline app/src/main/java/org/liberty/android/fantastischmemo/dao/CardDaoImpl.java]: we should probably have our variable names to be more meaningful. So change 'cs'

File: app/src/main/java/org/liberty/android/fantastischmemo/dao/CardDaoImpl.java [hunk 4]
```diff
@@ -858,5 +869,67 @@ private void maintainOrdinal() throws SQLException {
             throw new RuntimeException(e);
         }
     }
-}
+
+    public void shuffleCards(){
```
Comment (rekandiah) [inline app/src/main/java/org/liberty/android/fantastischmemo/dao/CardDaoImpl.java]: add Override annotation

File: app/src/main/java/org/liberty/android/fantastischmemo/dao/CardDaoImpl.java [hunk 5]
```diff
@@ -858,5 +869,67 @@ private void maintainOrdinal() throws SQLException {
...
+                    index++;
+                }
+                return null;
+            });
+        } catch (Exception e) {
+            throw new RuntimeException("An error occured while shuffling cards.", e);
+        }
+
+    }
+
```
Comment (rekandiah) [inline app/src/main/java/org/liberty/android/fantastischmemo/dao/CardDaoImpl.java]: add override annotation

File: app/src/main/java/org/liberty/android/fantastischmemo/dao/CardDaoImpl.java [hunk 6]
```diff
@@ -858,5 +869,67 @@ private void maintainOrdinal() throws SQLException {
...
+            //swap cards[i] and cards[newIndex]
+            Card temp = cs[i];
+            cs[i] = cs[newNum];
+            cs[newNum] = temp;
+        }
+
+        cards = Arrays.asList(cs);
+        try {
+            List<Card> finalCards = cards;
+            callBatchTasks((Callable<Void>) () -> {
```
Comment (rekandiah) [inline app/src/main/java/org/liberty/android/fantastischmemo/dao/CardDaoImpl.java]: this line is causing jenkins to fail. lambda expressions are not supported.

**Comment by rekandiah** [comment]
capture

File: app/src/main/java/org/liberty/android/fantastischmemo/dao/CardDaoImpl.java (hunk 4 above)
Comment (coreylevine) [reply app/src/main/java/org/liberty/android/fantastischmemo/dao/CardDaoImpl.java]: Why? It is not overriding any methods

File: app/src/main/java/org/liberty/android/fantastischmemo/dao/CardDaoImpl.java (hunk 5 above)
Comment (coreylevine) [reply app/src/main/java/org/liberty/android/fantastischmemo/dao/CardDaoImpl.java]: Why?

File: app/src/main/java/org/liberty/android/fantastischmemo/dao/CardDaoImpl.java (hunk 4 above)
Comment (rekandiah) [reply app/src/main/java/org/liberty/android/fantastischmemo/dao/CardDaoImpl.java]: @ override should be used for every method intended as an implementation of an interface as well. I think it's used for this case as well, you don't think so? If it's not consistent with all the other methods in the other impls then yeah don't add it (my bad)

**Comment by rekandiah** [comment]
blh

**Pull request merged**